config/routes.rb

```diff
@@ -25,7 +25,7 @@
   resources :sitemap, defaults: { format: :xml }, only: [:index, :show]
 
   # constraint to allow dots in id, also had to add needed format specifically
-  resources :solr_documents, id: /[^\/]+(?=\.html\z|\.json\z|\.xml\z)|[^\/]+/, only: [:show], path: '/catalog', controller: 'catalog' do
+  resources :solr_documents, id: /\d+[.,;:!"')\]]?/, only: [:show], path: '/catalog', controller: 'catalog' do
```

Comment:
this special punctuation regex character class could be held in a constant I guess 🤷 . probably fine as is